### PR TITLE
TypeSpecifier: Narrow `(int) $expr` like `$expr != 0`

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan-src/pull/3380#discussion_r1740085169

### CODE REVIEW

File: tests/PHPStan/Analyser/nsrt/narrow-bool-cast.php [hunk 1]
```diff
@@ -27,3 +27,13 @@ function doFoo(string $x, array $arr): void {
 	}
 	assertType('array{}|array{string}', $matches);
 }
+
+
+/** @param int<-5, 5> $x */
+function castInt($x) {
```
Comment (staabm) [inline tests/PHPStan/Analyser/nsrt/narrow-bool-cast.php]: I couldn't think of a better test-sample. 

I think testing type-inference will get more meaningful examples, after we understand loose comparisons better

**Comment by ondrejmirtes** [comment]
Conflict here

**Comment by staabm** [comment]
> Conflict here

resolved

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**